docs(linter): Update not_supported rules list with two stylistic react rules

connorshea opened the pull request:
These two rules will conflict with oxfmt's styling and should not be implemented by oxlint as native rules.

Docs for these two rules:

- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md
- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-tag-location.md

camc314 commented:
### Merge activity

* **Dec 30, 6:13 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 30, 6:13 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Update not_supported rules list with two stylistic react rules (#17504)

These two rules will conflict with oxfmt's styling and should not be implemented by oxlint as native rules.

Docs for these two rules:

- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md
- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-closing-tag-location.md
